src/components/ListItem.jsx

```diff
@@ -4,17 +4,79 @@ import React, { useState } from 'react';
 // LOCAL IMPORTS
 import './ListItem.css';
 import { updateItem } from '../api/firebase.js';
+import { calculateEstimate } from '@the-collab-lab/shopping-list-utils';
 
-export function ListItem({ name, listToken, dateLastPurchased, itemId }) {
+export function ListItem({
+	name,
+	listToken,
+	dateLastPurchased,
+	itemId,
+	totalPurchases,
+	dateCreated,
+	dateNextPurchased,
+}) {
+	const now = Date.now() / 1000; //current time in seconds
 	// SET STATES
 	const [isChecked, setIsChecked] = useState(
-		Date.now() / 1000 - dateLastPurchased?.seconds < 60 * 60 * 24,
+		now - dateLastPurchased?.seconds < 60 * 60 * 24,
 	);
 
+	// HELPER FUNCTIONS
+	const getDaysSinceLastTransaction = (dateLastPurchased, dateCreated) => {
+		const oneDay = 24 * 60 * 60; // hours * minutes * seconds
+		const lastTransactionDate = dateLastPurchased
+			? dateLastPurchased.seconds
+			: dateCreated.seconds; // Use the last purchased date if available, else use the created date
+
+		const daysSinceLastTransaction = Math.round(
+			Math.abs((now - lastTransactionDate) / oneDay),
+		);
+
+		return daysSinceLastTransaction;
+	};
+
+	// Helper fxn to calculate days between last and next purchased date
+	const getDaysBetweenLastAndNextPurchase = (
+		dateLastPurchased,
+		dateCreated,
+	) => {
+		const dateLastPurchasedSeconds = dateLastPurchased
+			? dateLastPurchased.seconds
+			: dateCreated.seconds;
+		const dateNextPurchasedSeconds = dateNextPurchased.seconds;
```

Comment:
This is confusing because it is not a parameter to this function but is still being used. If you extract this in the dates.js file it will require you to change it and help improve the clarity of intent.